CMakeLists.txt

```diff
@@ -2,6 +2,8 @@
 
 cmake_minimum_required(VERSION 2.8)
 
+set(CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS_DEBUG}   -ggdb")
```

Comment:
I guess for gcc ok , not sure about clang, vc14 will fail on this one for sure
Can you wrap it into macro please ?

if(WIN32)
else(WIN32)
    set(CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS_DEBUG}   -ggdb)
endif(WIN32)